lib/siblings/throttler/producer.ex

```diff
@@ -0,0 +1,43 @@
+defmodule Siblings.Throttler.Producer do
+  @moduledoc false
+  use GenStage
+
+  alias Siblings.Throttler
+
+  def start_link(initial \\ []),
+    do: GenStage.start_link(__MODULE__, initial)
+
+  @spec init([Throttler.throttlee()]) :: {:producer, [Throttler.throttlee()]}
+  @impl GenStage
+  def init(initial) when is_list(initial),
+    do: {:producer, normalize(nil, initial)}
+
+  @impl GenStage
+  def handle_demand(demand, items) when demand > 0 do
+    {head, tail} = Enum.split(items, demand)
+    {:noreply, head, tail}
+  end
+
+  @impl GenStage
+  def handle_call({:add, items}, from, state) when is_list(items),
+    do: {:noreply, [], state ++ normalize(from, items)}
+
+  def handle_call({:add, item}, from, state),
+    do: handle_call({:add, [item]}, from, state)
```

Comment:
By returning `{:noreply, _, _}` tuple from `handle_call/2` we pass through the current process’ event loop, but the initial `call/2` stays blocked until `GenStage.reply/2` with the respective `from` would be called.